sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubClientBuilder.java

```diff
@@ -282,14 +282,19 @@ public WebPubSubAsyncServiceClient buildAsyncClient() {
 
         if (endpoint == null && credential == null) {
             final Map<String, String> csParams = parseConnectionString(connectionString);
+            if (!csParams.containsKey("endpoint") && !csParams.containsKey("accesskey")) {
+                logger.logThrowableAsError(new IllegalArgumentException(
+                    "Connection string does not contain required 'endpoint' and 'accesskey' values"));
+            }
+
             final String accessKey = csParams.get("accesskey");
 
             this.credential = new AzureKeyCredential(accessKey);
             this.endpoint = csParams.get("endpoint");
 
-            if (!csParams.containsKey("endpoint") && !csParams.containsKey("accesskey")) {
-                logger.logThrowableAsError(new IllegalArgumentException(
-                    "Connection string does not contain required 'endpoint' and 'accesskey' values"));
+            String port = csParams.get("port");
+            if (!CoreUtils.isNullOrEmpty(port)) {
+                this.endpoint = this.endpoint + ":" + port;
```

Comment:
endpoint is a URL, it is possible that it also contains subpath, can we only override port value for the endpoint?